homeassistant/components/vesync/config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Config flow utilities."""
+import logging
+from collections import OrderedDict
+import voluptuous as vol
+from pyvesync import VeSync
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return already configured instances."""
+    return hass.config_entries.async_entries(DOMAIN)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class VeSyncFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+        self._username = None
+        self._password = None
+        self._time_zone = None
+        self.data_schema = OrderedDict()
+        self.data_schema[vol.Required(CONF_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+
+    async def _show_form(self, errors=None):
```

Comment:
This can be a callback instead since we don't need to await inside.

Define it as a regular method and decorate it with `@callback` imported from core.py.

We prefer to use callbacks over coroutines since calling coroutines is slower.